### PR TITLE
feat(billing): pricing page redesign — multi-mode + auto-savings + sectioned features + FAQ

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

Refactor of the public pricing page (`/pricing`) on the **devkit Vue stack**. Plan: `infra/docs/superpowers/plans/2026-05-08-pricing-page-redesign.md`.

**What ships:**
- **3 pricing modes** (driven by static-content `pricingMode`): `subscription` / `packs` / `both-tabs`. Legacy fallback preserved (server `meterMode` + packs presence) for downstream projects that haven't opted in.
- **Auto-calculated annual savings**: `computeAnnualSavingsPct(plan)` per plan + `computeMaxAnnualSavingsPct(plans)` for the toggle. Toggle now displays "Save up to X%" dynamically; each card shows an inline "Save X% with annual" chip when annual is selected. No more static "save 17%" copy.
- **Configurable feature sections**: new `featureSections` schema with optional `title`, `inheritsFrom` (renders "Everything in {parent}, plus"), and per-item `icon` / `tooltip` / `highlight`. Backward-compat: legacy flat `features: [{text, included}]` continues to render unchanged when `featureSections` is empty/absent.
- **FAQ accordion** with embedded `schema.org` `FAQPage` JSON-LD for SEO / AI Overview.
- **`usePricing` composable** centralises mode + plans + packs + faqs + savings resolution. View shrunk from 419 → 310 lines.

## Visual QA setup (READ THIS BEFORE REVIEWING)

The devkit `static-content.js` is **preloaded with demo data** (3 demo packs, 3 demo FAQs, `pricingMode='both-tabs'`) so the dev server shows the full multi-mode UX immediately. Run locally to see the redesign:

```bash
cd /Users/pierrebrisorgueil/Documents/Dev/github/pierreb-devkit/Vue/.worktrees/pricing-redesign
npm run dev
# visit http://localhost:<port>/pricing
```

Try:
- Toggle Monthly/Annual → "Save up to X%" chip in toggle + per-card chip
- Click "Units" tab → demo packs grid
- Scroll down → FAQ accordion with 3 entries
- Inspect page source → schema.org JSON-LD `FAQPage` markup present

Downstream projects (Trawl, Comes, etc.) own their own `static-content.js` (it is not synced via /update-stack), so this devkit demo data does NOT leak. Each downstream project will populate its own packs / faqs / pricingMode when adopting the redesign.

## Test plan

- [x] Unit: 1562 tests, all passing
- [x] Lint: clean
- [x] Audit prod: 0 vulnerabilities
- [x] Backward-compat: legacy flat `features` plans still render correctly (existing pricingCard tests preserved + extended)
- [x] Helper math: `pricingMath.js` 15 cases covering edge cases (free plan, missing prices, zero discount, max across plans, mode resolution legacy fallback chain, unknown mode throws)
- [ ] Visual QA via dev server (mandatory before merge — see setup above)

## Files

- NEW: `src/modules/billing/lib/pricingMath.js` + tests
- NEW: `src/modules/billing/composables/billing.usePricing.js` + tests
- NEW: `src/modules/billing/components/billing.pricingFeatureSection.component.vue` + tests
- NEW: `src/modules/billing/components/billing.pricingFAQ.component.vue` + tests
- MODIFY: `src/modules/billing/components/billing.pricingCard.component.vue` + tests (backward-compat preserved)
- MODIFY: `src/modules/billing/components/billing.pricingToggle.component.vue` + tests
- MODIFY: `src/modules/billing/views/billing.pricing.view.vue` + tests (419 → 310 lines, mode-aware orchestrator)
- MODIFY: `src/modules/billing/config/billing.static-content.js` (extended schema + demo data for QA)
- MODIFY: `src/modules/billing/lang/en.js` + `fr.js` (4 new keys: `pricingToggle.saveUpTo`, `pricingToggle.saveAnnuallyDynamic`, `pricingCard.saveAnnual`, `pricingFeatureSection.everythingIn`, `pricing.faq.title`)

## Release Notes

* **New Features**
  * Annual savings percentages now displayed on pricing cards when applicable
  * Pricing plans display organized feature sections showing included benefits and tier differences
  * FAQ section added to pricing page
  * Pricing page supports tabbed navigation for subscription and usage-based billing modes

* **Style**
  * Enhanced visual layout of pricing cards with improved badge positioning
  * Feature lists now show icons and highlighting for key benefits

### CODE REVIEW

File: src/modules/billing/views/billing.pricing.view.vue [hunk 1]
```diff
@@ -344,80 +253,48 @@ export default {
...
       const targetTier = this.planTierOrder.indexOf(planId);
       const currentTier = this.planTierOrder.indexOf(this.currentPlanId);
       if (targetTier !== -1 && currentTier !== -1 && targetTier < currentTier) {
         this.pendingDowngrade = { planId, priceId };
         this.downgradeDialog = true;
         return;
       }
-
       await this.proceedCheckout({ planId, priceId });
     },
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply src/modules/billing/views/billing.pricing.view.vue:347]: Free-plan clicks returning early is intentional — downgrading to free requires a subscription cancellation flow (calling the billing portal or a cancel endpoint), not a Stripe checkout. `proceedCheckout` only handles paid plan checkouts and would fail with a null priceId. Implementing a full cancel-to-free flow is out of scope for this redesign PR. Deferring to a future billing-portal integration story. The CTA button on the free card is only shown when the user is not already on free (see pricingCard logic), so there is no confusion for free-tier users.

File: src/modules/billing/composables/billing.usePricing.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { useAuthStore } from '../../auth/stores/auth.store.js';
+import { useBillingStore } from '../stores/billing.store.js';
+import {
+  pricingMode as staticPricingMode,
+  plans as staticPlans,
+  packs as staticPacks,
+  faqs as staticFaqs,
+} from '../config/billing.static-content.js';
+import { computeMaxAnnualSavingsPct, resolvePricingMode } from '../lib/pricingMath.js';
+
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply src/modules/billing/composables/billing.usePricing.js:24]: False positive — the devkit's `billing.static-content.js` is not synced by `/update-stack`. Downstream projects own their own copy and are NOT affected by devkit template changes. The `pricingMode`, `packs`, and `faqs` exports were added as part of this PR (see the modified `billing.static-content.js`). Any downstream project that hasn't adopted the redesign keeps its existing static-content.js untouched.
